warehouse/manage/views.py

```diff
@@ -729,13 +731,25 @@ def delete_project_role(project, request):
     permission="manage:project",
 )
 def manage_project_history(project, request):
-    journals = (
+    try:
+        page_num = int(request.params.get("page", 1))
+    except ValueError:
+        raise HTTPBadRequest("'page' must be an integer.") from None
+
+    journals_query = (
         request.db.query(JournalEntry)
         .options(joinedload("submitted_by"))
         .filter(JournalEntry.name == project.name)
         .order_by(JournalEntry.submitted_date.desc(), JournalEntry.id.desc())
-        .all()
     )
+
+    journals = SQLAlchemyORMPage(
+        journals_query,
+        page=page_num,
+        items_per_page=25,
```

Comment:
It may be convenient for testing to have this as a setting, possibly opening the door for a nice dropdown in the pagination controls in the future

Reply:
I would be happy to make `items_per_page` a setting, but I think I need some guidance on how I should go about doing that.  Thanks for reviewing and for your feedback!

Reply:
On second thought it might work better as a request paramenter read by the view, but only makes sense if we were to implement the feature in the frontend so for now it's fine as it is. 👍